### ISSUE TITLE
Reference Brackets Assigned by Reference Brackets Do Not Enforce Spaces.

### ISSUE THREAD

**Issue by joshuapinter** [bbatsov/rubocop#5806]
To be clear, whether you have `EnforcedStyle: no_space` or `EnforcedStyle: space`, the first Reference Bracket is completely ignored when assigning from another Reference Bracket.

## Expected behavior

Rubocop to register an offence when there is no space in the reference bracket being assigned:

**`EnforcedStyle: no_space`**
```ruby
a[ "foo"] = b["something"]
  ^ Do not use space inside reference brackets.
```

**`EnforcedStyle: space`**
```ruby
a["foo"] = b[ "something" ]
 ^ Use space inside reference brackets.
```

## Actual behavior

No offence was registered in either test case.

## Steps to reproduce the problem

Test the above lines with Rubocop, either enforcing spaces or not enforcing spaces.

## RuboCop version

```bash
$ bundle exec rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.2.0 x86_64-darwin16)
```

**Comment by joshuapinter** [comment]
I have forked, branched and added two failing tests, one for `EnforcedStyle: no_space` and `EnforcedStyle: space`.

**Comment by joshuapinter** [comment]
I forked and fixed this issue in Pull Request #5807.

**Comment by garettarrowood** [comment]
Resolved in #5807 . This issue can/should be closed.

**Comment by joshuapinter** [comment]
@garettarrowood Agreed! Must not have put "Fixes #5806" in the PR. Closing now...
